### PR TITLE
Support dotted notation in attribute path

### PR DESCRIPTION
**Pull request by baijum**
Fixes https://jira.coreos.com/browse/ODC-1243

### CODE REVIEW

File: pkg/controller/servicebindingrequest/retriever.go [hunk 1]
```diff
@@ -28,6 +28,24 @@ const (
 	secretPrefix  = "urn:alm:descriptor:servicebindingrequest:env:object:secret"
 )
 
+func getNestedValue(key string, sectionMap interface{}) (string, error) {
```
Comment (otaviof) [inline pkg/controller/servicebindingrequest/retriever.go]: I would expect to have `getNestedValue` as parte of the `Retriever` type. Is there a reason for standalone func?

File: pkg/controller/servicebindingrequest/retriever.go [hunk 2]
```diff
@@ -28,6 +28,24 @@ const (
...
+	var value interface{}
+	var exists bool
+	if !strings.Contains(key, ".") {
+		value, exists = sectionMap.(map[string]interface{})[key]
```
Comment (otaviof) [inline pkg/controller/servicebindingrequest/retriever.go]: You can just use inline declaration here (`:=`), since after this `if` you declare them again (line 42).

File: pkg/controller/servicebindingrequest/retriever.go [hunk 3]
```diff
@@ -28,6 +28,24 @@ const (
...
+		if !exists {
+			return "", fmt.Errorf("Can't find key '%s'", key)
+		}
+		return fmt.Sprintf("%v", value), nil
+	}
+	attrs := strings.SplitN(key, ".", 2)
```
Comment (otaviof) [inline pkg/controller/servicebindingrequest/retriever.go]: Do you need to check for `len()`, before doing the `SplitN`?

File: pkg/controller/servicebindingrequest/retriever.go (hunk 1 above)
Comment (baijum) [reply pkg/controller/servicebindingrequest/retriever.go]: Retriever object is not accessed anywhere within the function. So, the standalone function would be fine.

File: pkg/controller/servicebindingrequest/retriever.go (hunk 3 above)
Comment (baijum) [reply pkg/controller/servicebindingrequest/retriever.go]: That's not really required. This function will not return any error.

File: pkg/controller/servicebindingrequest/retriever.go (hunk 2 above)
Comment (baijum) [reply pkg/controller/servicebindingrequest/retriever.go]: Done. 7351fde1f36e5fe43f8847745714b967842bd9dd

File: pkg/controller/servicebindingrequest/retriever_test.go [hunk 4]
```diff
@@ -102,3 +102,44 @@ func TestRetriever(t *testing.T) {
...
+	}
+
+	dbSecret := mocks.SecretMock(ns, "db-credentials")
+	objs := []runtime.Object{&dbSecret}
+	fakeClient := fake.NewFakeClient(objs...)
+
+	retriever = NewRetriever(context.TODO(), fakeClient, plan)
+	require.NotNil(t, retriever)
+
+	t.Run("Second level", func(t *testing.T) {
```
Comment (ldimaggi) [inline pkg/controller/servicebindingrequest/retriever_test.go]: Is it worth adding an edge case here - where the path does not include a "." char?

Or - maybe a test of an invalid path (that should raise an error) such as "first..second"?

Comment (baijum) [reply pkg/controller/servicebindingrequest/retriever_test.go]: There is a test without any dot in line 50.

Added test for two dot scenario: b0f80a0c68a422e9d941f3b62f2fb9eb48c8f93a

File: pkg/controller/servicebindingrequest/retriever.go (hunk 1 above)
Comment (otaviof) [reply pkg/controller/servicebindingrequest/retriever.go]: Indeed it does not access, but it's also not used anywhere else. Furthermore, is not ready to be used anywhere else, since it belongs to `Retriever` type. Since it does belong to Retriever and is only ready to be used in that context, I think it either deserves a standalone type, or join Retriever.
But leaving a global method with local scope is not something we should do in this case.

Comment (otaviof) [reply pkg/controller/servicebindingrequest/retriever.go]: By the way, we are also missing a doc-comment on this function, please.

Comment (baijum) [reply pkg/controller/servicebindingrequest/retriever.go]: Done. 991bb922eb12a0d1658ecc69521e64b51d12bbfc

Comment (otaviof) [reply pkg/controller/servicebindingrequest/retriever.go]: Thanks :-)
